<ul>
<li><a href=#question7>7. Ode to n</a></li>
<li><a href=#question8>8. Calculate Percentages</a></li>
<li><a href=#question9>9. Common Decade</a></li>
<li><a href=#question10>10. How Old?</a></li>
<li><a href=#question11>11. Really Old!</a></li>
<li><a href=#question12>12. Birth Year Histogram</a></li>
</ul>
</ul>

<a id='question7'></a>

### 7. Ode to n

In [5]:
import pandas as pd

birth_years = pd.read_csv('birth_year.csv')
birth_years

Birth year  Frequency
0  1911-1920          1
1  1921-1930          2
2  1931-1940         10
3  1941-1950         59
4  1951-1960        206
5  1961-1970        454
6  1971-1980       1323
7  1981-1990       3986
8  1991-2000       1851
9  2001-2010          9

In [9]:
birth_years['Frequency'].sum()

7901

<a id='question8'></a>

### 8. Calculate the percentage of students in the indicated birth year bins.

In [13]:
# Add a column named 'Percentage' which value is based on the Frequency divided by the sum of frequencies(n) and then
# multiplied by 100. 

birth_years['Percentage'] = (birth_years['Frequency']/birth_years['Frequency'].sum())*100
birth_years

Birth year  Frequency  Percentage
0  1911-1920          1    0.012657
1  1921-1930          2    0.025313
2  1931-1940         10    0.126566
3  1941-1950         59    0.746741
4  1951-1960        206    2.607265
5  1961-1970        454    5.746108
6  1971-1980       1323   16.744716
7  1981-1990       3986   50.449310
8  1991-2000       1851   23.427414
9  2001-2010          9    0.113910

<a id='question9'></a>

### 9. Common Decade

What is the most common decade in which this sample of Udacity students was born?

In [20]:
# Select the row where the Frequency is the maximum. (selecht the id of the row and then return the row by using .iloc)
# http://pandas.pydata.org/pandas-docs/stable/10min.html#getting

birth_years.iloc[birth_years['Frequency'].idxmax()]

Birth year    1981-1990
Frequency          3986
Percentage     50.44931
Name: 7, dtype: object

<a id='question10'></a>

### 10. How Old?

Approximately what percent of students were born before 1975?

In [23]:
bornbefore = 0
for i in range(0,6):
    bornbefore = bornbefore + birth_years['Percentage'][i]
bornbefore
    

9.2646500442981896

Note that since 1975 falls in the middle of the 1971-1980 bin, any answer between 9.26% and 26% could be the possible true percentage.

<a id='question11'></a>

### 11. Really Old!

Is the oldest student at least 90 years old?

- Yes
- No
- Cannot be determined

<b>Answer:</b> Yes

<a id='question12'></a>

### 12. Birth Year Histogram

Whats is the bin size of each bar in this histogram?

In [32]:
#Unable to plot this as shown in the question because we don't have the raw data.

import matplotlib.pyplot as plt